src/app/modules/core/components/user-profile/user-profile.component.ts

```diff
@@ -1,14 +1,152 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { DATE_FORMATS } from '../../constants/appConstants';
+import { StateCity } from '../../interfaces/state-city';
+import { UserProfile } from '../../interfaces/user-profile';
+import { MockService } from '../../services/mock.service';
+import { validateEmail, validateMobileNo } from '../../utilities/helper';
 
 @Component({
   selector: 'app-user-profile',
   templateUrl: 'user-profile.component.html',
   styleUrls: ['user-profile.component.css'],
 })
-
 export class UserProfileComponent implements OnInit {
-  constructor() {}
+  public screenScope: { profile: boolean; contactList: boolean };
+  public userProfileForm: FormGroup;
+  public filteredStates: Array<string>;
+  public filteredCities: Array<string>;
+  public stateFilter;
+  public cityFilter;
+  public masterStateCityList: Array<StateCity>;
+  public minDate: Date;
+  public maxDate: Date;
+
+  constructor(
+    private snackBar: MatSnackBar,
+    private formBuilder: FormBuilder,
+    private mockService: MockService
+  ) {
+    this.screenScope = { profile: false, contactList: true };
+    const currentYear = new Date().getFullYear();
+    this.minDate = new Date(currentYear - 20, 0, 1);
+    this.maxDate = new Date(currentYear + 1, 11, 31);
+  }
+
+  ngOnInit() {
+    this.mockService.getCities().subscribe((list) => {
+      if (Array.isArray(list) && list.length > 0) {
+        this.masterStateCityList = list;
+        this.filteredStates = Array.from(
+          new Set(
+            list.map((stateCity: StateCity) => {
+              return stateCity.state;
+            })
+          )
+        );
+      }
+    });
+    this.setDefaultValuesInForm();
+  }
+
+  private setDefaultValuesInForm() {
+    let userProfile: UserProfile;
+    let existingProfile: UserProfile = JSON.parse(
+      localStorage.getItem('myProfile')
+    );
+    let compositeField = localStorage.getItem('userId');
+    userProfile = {
+      fullName:
+        existingProfile && existingProfile.fullName
+          ? existingProfile.fullName
+          : '',
+      emailId:
+        existingProfile && existingProfile.emailId
+          ? existingProfile.emailId
+          : compositeField && validateEmail(compositeField)
```

Comment:
why do we need to validate email here? Its already validated and saved to local storage right? Same question for phone number.

Reply:
since compositeField can be an email or phone number and we need to repopulate the same data in the form so we first need to verify wether the composite field actually a mobile number or an email id